h2o-k8s/src/main/java/water/k8s/api/KubernetesRestApi.java

```diff
@@ -0,0 +1,77 @@
+package water.k8s.api;
+
+import fi.iki.elonen.NanoHTTPD;
+import fi.iki.elonen.router.RouterNanoHTTPD;
+import water.k8s.probe.KubernetesLeaderNodeProbeHandler;
+
+import java.io.IOException;
+
+/**
+ * This class represents a tiny (memory, cpu, dependencies) self-contained API only for Kubernetes,
+ * running separately on localhost on a specified port.
+ * When the Kubernetes extension is starting and attempting to form an H2O cluster,
+ * H2O's REST API is yet not fully initialized, as its startup relies on configuration obtained
+ * during the clustering phase. However, REST API services (readiness probe, liveness probe, startup probe)
+ * are required to be available since the very start of H2O K8S extension. Therefore, a separate REST API running
+ * on a distinct port is spawned.
+ * <p>
+ */
+public class KubernetesRestApi extends RouterNanoHTTPD implements AutoCloseable {
+
+    /**
+     * Default port to bind to / listen on.
+     */
+    private static final int DEFAULT_PORT = 8080;
+    public static final String KUBERNETES_REST_API_PORT_KEY = "H2O_K8S_API_PORT";
+
+
+    /**
+     * Creates, but not starts Kubernetes REST API. To start the REST API, please use
+     * one of the start methods available.
+     * <p>
+     * The REST API is bound to a default port of 8080, unless specified otherwise by H2O_K8S_API_PORT environment
+     * variable.
+     */
+    public KubernetesRestApi() {
+        super(getPort());
+        addMappings();
+    }
+
+    private static int getPort() {
+        final String customKubernetesPort = System.getenv(KUBERNETES_REST_API_PORT_KEY);
```

Comment:
why not use the standard h2o property prefix?

Reply:
`H2O_K8S_API_PORT` hides under the `KUBERNETES_REST_API_PORT_KEY` static field. This is in-line with the rest of H2O K8S properties. A simple `H2O` prefix is used for K8S to save space in YAML files. It's easy and short and doesn't make heavily indented lines any longer with unnecessary prefixes like our system property prefix `public static final String SYSTEM_PROP_PREFIX = "sys.ai.h2o.";`

This style is also in-line with what K8S uses: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/

Let's compare the two styles:

```
          - name: H2O_KUBERNETES_SERVICE_DNS
            value: h2o-service.h2o-statefulset.svc.cluster.local
          - name: H2O_NODE_LOOKUP_TIMEOUT
            value: '180'
          - name: H2O_NODE_EXPECTED_COUNT
            value: '3'
          - name: H2O_K8S_API_PORT
            value: '8081'
```

SYS_AI prefix: 
```
          - name: SYS_AI_H2O_KUBERNETES_SERVICE_DNS
            value: h2o-service.h2o-statefulset.svc.cluster.local
          - name: SYS_AI_H2O_NODE_LOOKUP_TIMEOUT
            value: '180'
          - name: SYS_AI_H2O_NODE_EXPECTED_COUNT
            value: '3'
          - name: SYS_AI_H2O_K8S_API_PORT
            value: '8081'
```

From user's POV, what SYS_AI even means ? It's long and unnecessary. H2O properties are prefixed with H2O. This is simplest and most user friendly and meaningful.

Reply:
I'm not 100% against it, yet I don't think our K8S env vars need to follow the same long pattern as our sys props.

Reply:
Actually now, when I'm looking at it, I have H2O_KUBERNETES in one field and H2O_K8S in another. This must be aligned to H2O_KUBERNETES.